### PR TITLE
Additional plugins to capture debug data to BMC dump

### PR DESCRIPTION
**Pull request by devenrao**
Additional dump plugins to be added to BMC dump generation
CQ: SW550084

https://gerrit.openbmc-project.xyz/c/openbmc/phosphor-debug-collector/+/53284
https://gerrit.openbmc-project.xyz/c/openbmc/phosphor-debug-collector/+/53285
https://gerrit.openbmc-project.xyz/c/openbmc/phosphor-debug-collector/+/53291
https://gerrit.openbmc-project.xyz/c/openbmc/phosphor-debug-collector/+/53292
https://gerrit.openbmc-project.xyz/c/openbmc/phosphor-debug-collector/+/53293

### CODE REVIEW

File: tools/dreport.d/plugins.d/journalpid
Comment (RameshIyyar) [inline tools/dreport.d/plugins.d/journalpid:3]: Cannot we use https://github.com/openbmc/phosphor-debug-collector/blob/master/tools/dreport.d/plugins.d/journalpretty? looks like we are collecting full journal traces for all types of the BMC dump, right? any specific reason?

**Comment by RameshIyyar** [comment]
Please include upstream Gerrit links for the same

File: tools/dreport.d/plugins.d/journalpid
Comment (devenrao) [reply tools/dreport.d/plugins.d/journalpid:3]: as per discussion in SOS, and in openbmc-general

Comment (RameshIyyar) [reply tools/dreport.d/plugins.d/journalpid:3]: > Journal logs in core dumps and internal failures: Just grab the full journal

I think you are collecting full traces as discussed but, both plugins collect full traces so, cannot use only one by adding the config `13` into https://github.com/ibm-openbmc/phosphor-debug-collector/blob/1020/tools/dreport.d/plugins.d/journalpretty#L3?

and also we will miss entries if the application is added because we are using the `--no-pager`.

File: tools/dreport.d/plugins.d/bmcweb
Comment (RameshIyyar) [inline tools/dreport.d/plugins.d/bmcweb:9]: any specific reason?

**Review by rsnair — COMMENTED** [review]
We only need to collect /home/root/bmcweb_current_session_snapshot.json

File: tools/dreport.d/plugins.d/bmcweb
Comment (rsnair) [reply tools/dreport.d/plugins.d/bmcweb:9]: > any specific reason?

This file aid in debug for certain field issues by collecting information on the open sessions and event subscribers for bmcweb

Comment (RameshIyyar) [reply tools/dreport.d/plugins.d/bmcweb:9]: Sorry for the confusion why there is a check to make sure the bmcweb application is installed or not?  Won't we support the Redfish in certain OpenBMC based OpenPOWER systems?

Comment (devenrao) [reply tools/dreport.d/plugins.d/bmcweb:9]: this check is done in general as with bitbake we have the capability to include/not include a pacakge or start/don't start a service.

File: tools/dreport.d/plugins.d/journalpid
Comment (ojayanth) [inline tools/dreport.d/plugins.d/journalpid:1]: Enable journalpretty for elog/core is going to meet the requirement right? incase you are not using journal-pid you can remove from down stream. 
Here yu are keepin two plugins with same name. that is not good.

File: tools/dreport.d/plugins.d/bmcweb
Comment (ojayanth) [inline tools/dreport.d/plugins.d/bmcweb:1]: Need BMCweb maintainer approval before merging this into ghe.

Comment (rsnair) [reply tools/dreport.d/plugins.d/bmcweb:1]: @sunharis @gtmills Please check on this. This was the additional sessions & event data that was agreed to be captured as part of BMC dump for downstream. bmcweb changes for this was merged a while back.

**Review by sunharis — APPROVED** [review]

File: tools/dreport.d/plugins.d/journalpid
Comment (devenrao) [reply tools/dreport.d/plugins.d/journalpid:1]: ok

**Review by RameshIyyar — APPROVED** [review]
LGTM for the GHE- some minor comments posted in the upstream patches but, no functional impacts.

**Review by ojayanth — APPROVED** [review]

**Pull request merged**
